### PR TITLE
#525 - Create CR Table Clear Filter Button

### LINKED ISSUE #525: Change Requests - Clear Current Filters Button
## Description

A new button needs to be added to the filter component on the change request table page. This would allow users to clear all filters with the click of a button, instead of needing to deselect them manually one-by-one.

## Background

The filters section on the change request page which displays all the current change requests has several filters you can apply to limit which change requests are displayed in the table. Currently, the only ways to reset these filters are to manually do them, go to another page and come back, or reload the application and then renavigate to the change request page; all of these are somewhat cumbersome.

## Proposed Solution

- [ ] Create a button "Clear" on the filter component on the change request table page according to the mocks below
  - [ ] change request table filter already has a flexible [`sendDataToParent` function to utilize](https://github.com/Northeastern-Electric-Racing/PM-Dashboard-v2/blob/daacad86adfe70f1230f8e9e5e7d2af1eca554b0/src/components/change-requests/change-requests-table/change-requests-table.tsx#L140)
  - [ ] add an adjacent [new button that has the same `onClick` except](https://github.com/Northeastern-Electric-Racing/PM-Dashboard-v2/blob/daacad86adfe70f1230f8e9e5e7d2af1eca554b0/src/components/change-requests/change-requests-table/change-requests-filter/change-requests-filter.tsx#L142-L149) instead of passing local state variables, pass in the state variable default values
  - [ ] also be sure to reset the filter's local state variables
  - [ ] test the new functionality

## Expected Behavior

- [ ] Clicking on the button will clear all the filters you currently have set on the change request table
  - [ ] All the buttons in the filter will be unchecked
  - [ ] All the drop-down menus in the filter will be reset to the blank state
- [ ] Redisplay all change request rows

## Mocks

This is a mockup after specified filters were applied

image

This is a mockup after the `clear` button was clicked

image

### PR DESCRIPTION
**Pull request by acd124**
## Changes

Created a clear button to reset all filters for the Change Request table.

## Test Cases

- Make sure clear button appears
- Make sure the clear button does not change when pressed
- Make sure clear resets parent state

## Screenshots (if applicable)

Screen Shot 2022-03-24 at 6 30 23 PM

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://github.com/Northeastern-Electric-Racing/PM-Dashboard-v2/blob/main/docs/ContributorGuide.md) and reach out to your squad if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors
- [x] No newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (if applicable)
- [x] Remove any not-applicable sections
- [x] Assign the PR to yourself
- [x] PR is linked to the ticket
- [x] No `package-lock.json` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack

Closes #525

### CODE REVIEW

File: src/components/change-requests/change-requests-table/change-requests-filter/change-requests-filter.module.css [hunk 1]
```diff
@@ -26,12 +26,21 @@
   transform: translateY(-50%);
 }
 
-.applyButton {
-  float: right;
+.applyButton,
+.clearButton {
```
Comment (jamescd18) [inline src/components/change-requests/change-requests-table/change-requests-filter/change-requests-filter.module.css:30]: nice use of CSS

**Review by jamescd18 — APPROVED** [review]
Looks good to me and looks like it behaves correctly in local dev! Awesome job!

File: src/components/change-requests/change-requests-table/change-requests-filter/change-requests-filter.module.css [hunk 2]
```diff
@@ -26,12 +26,21 @@
...
 }
 
-.applyButton:focus {
+.applyButton {
+  float: right;
+}
+
+.clearButton {
+  float: left;
+}
```
Comment (jpwaves) [inline src/components/change-requests/change-requests-table/change-requests-filter/change-requests-filter.module.css:41]: You can eliminate the use of floating elements by utilizing flexbox. You'll need to create a container to put both the buttons in. When you create the container element, add a class onto it that has `display: flex`, `align-items: center`, and `justify-content: space-between` (I think these are all the right properties but I might've messed up some dashes or capitalization) and that will do the same thing as your floats but without taking those elements out of the normal HTML flow. You may need to set the width of the container to `100%` or something like that to get the spacing of the buttons correct.

File: src/components/change-requests/change-requests-table/change-requests-filter/change-requests-filter.tsx [hunk 3]
```diff
@@ -141,11 +142,22 @@ const ChangeRequestsFilter: React.FC<FilterFieldStateProps> = ({
...
+              setImpact([]);
+              setReason('');
+              setState([]);
+              setImplemented('');
+              update('', [], '', [], '');
             }}
           >
-            Apply
+            Clear
           </Button>
```
Comment (jpwaves) [inline src/components/change-requests/change-requests-table/change-requests-filter/change-requests-filter.tsx:161]: You can put these 2 buttons inside a `div` or `Form.Group` and then add on the class that has all the flex box properties

**Review by jpwaves — CHANGES_REQUESTED** [review]
Looks mostly good, only thing is a comment about the usage of `float` for aligning and positioning of the buttons

**Pull request merged**